index.js

```diff
@@ -343,18 +353,31 @@ function processStyle(glStyle, map, baseUrl, host, path, accessToken) {
                     tileGrid.getResolution(tileGrid.getMinZoom()));
                 }
                 unByKey(key);
+              } else if (tilejson.getState() == 'error') {
+                layer.setSource(undefined);
+                unByKey(key);
               }
             });
             return layer;
           })();
         } else if (glSource.type == 'raster') {
-          let source;
+          layer = new TileLayer({
+            visible: glLayer.layout ? glLayer.layout.visibility !== 'none' : true
+          });
+          let source = undefined;
           if (!glSource.tiles) {
             source = new TileJSON({
               transition: transition,
               url: url,
               crossOrigin: 'anonymous'
             });
+            const key = source.on('change', function() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define this function outside of a loop. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sq.boundlessgeo.io/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)